point_pattern.py

```diff
@@ -119,8 +141,27 @@ def average_nearest_neighbor_distance(points):
      Measure of Spatial Relationships in Populations. Ecology. 35(4)
      p. 445-453.
     """
-    mean_d = 0
 
+    #d_i is the set of all of the distances between i and it's closest neighbor.
+    #then, between all of those distances, you divide by the number of points.
+
+    #find the distance between a point i and every other point
+    shDistL =[]       #list of shortest distances
+    for point in points:
+        shortestDistance = 9999999999
+        for dpoint in points:
+            if point != dpoint:
```

Comment:
Check out `enumerate()`.  The above logic works, until the points are coincident.